tests/filters/flattenComponents_test.py

```diff
@@ -0,0 +1,144 @@
+                    ('lineTo', ((300, 300),)),
+                    ('lineTo', ((0, 300),)),
+                    ('closePath', ()),
+                ],
+                'anchors': [
+                    (100, 200, 'top'),
+                    (100, -200, 'bottom'),
+                ],
+            },
+            {
+                'name': 'b',
+                'width': 350,
+                'outline': [
+                    ('addComponent', ('a', (1, 0, 0, 1, 0, 0))),
+                ],
+                'anchors': [
+                    (100, 200, 'top'),
+                    (100, -200, 'bottom'),
+                ],
+            },
+            {
+                'name': 'c',
+                'width': 350,
+                'outline': [
+                    ('addComponent', ('b', (1, 0, 0, 1, 0, 0))),
+                ],
+                'anchors': [
+                    (100, 200, 'top'),
+                    (100, -200, 'bottom'),
+                ],
+            },
+            {
+                'name': 'd',
+                'width': 700,
+                'outline': [
+                    ('addComponent', ('a', (1, 0, 0, 1, 0, 0))),
+                    ('addComponent', ('b', (1, 0, 0, 1, 350, 0))),
+                    ('addComponent', ('c', (1, 0, 0, 1, 700, 0)))
+                ],
+                'anchors': [
+                    (100, 200, 'top_1'),
+                    (450, 200, 'top_2'),
+                    (800, 200, 'top_3'),
+                    (100, -200, 'bottom_1'),
+                    (450, -200, 'bottom_2'),
+                    (800, -200, 'bottom_3'),
+                ],
+            },
+        ],
+    }
+])
+def font(request):
+    font = defcon.Font()
+    for param in request.param['glyphs']:
+        glyph = font.newGlyph(param['name'])
+        glyph.width = param.get('width', 0)
+        pen = glyph.getPen()
+        for operator, operands in param.get('outline', []):
+            getattr(pen, operator)(*operands)
+        for x, y, name in param.get('anchors', []):
+            glyph.appendAnchor(dict(x=x, y=y, name=name))
+    return font
+
+
+class FlattenComponentsFilterTest(object):
+
+    def test_empty_glyph(self, font):
+        philter = FlattenComponentsFilter(include={'space'})
+        assert not philter(font)
+
+    def test_contour_glyph(self, font):
+        philter = FlattenComponentsFilter(include={'a'})
+        assert not philter(font)
+
+    def test_component_glyph(self, font):
+        philter = FlattenComponentsFilter(include={'b'})
+        assert not philter(font)
+
+    def test_nested_components_glyph(self, font):
+        philter = FlattenComponentsFilter(include={'c'})
+        modified = philter(font)
+        assert modified == set(['c'])
+        assert (
+            [(c.baseGlyph, c.transformation) for c in font['c'].components] ==
+            [('a', (1, 0, 0, 1, 0, 0))]
+        )
+        assert (
+            [(a.name, a.x, a.y) for a in font['c'].anchors] ==
```

Comment:
what's the point of asserting the anchors here? The filter is not supposed to modify these, isn't it?

Reply:
OK